MAPL 2 GNU Build Broken

Per the nightly tests:
```
/discover/nobackup/mathomp4/SystemTests/builds/AGCM_GNUMAPLDEV/CURRENT/GEOSgcm/src/Shared/@MAPL/MAPL_Base/MAPL_ExtDataGridCompMod.F90:2048:43-43:

         type(Variable), pointer :: variable
                                           1
Error: The type 'variable' cannot be host associated at (1) because it is blocked by an incompatible object of the same name declared at (2)
make[2]: *** [src/Shared/@MAPL/MAPL_Base/CMakeFiles/MAPL_Base.dir/MAPL_ExtDataGridCompMod.F90.o] Error 1
make[1]: *** [src/Shared/@MAPL/MAPL_Base/CMakeFiles/MAPL_Base.dir/all] Error 2
make: *** [all] Error 2
```
We have seen this before. The issue is GNU doesn't like having the variable named the same as the type. The usual fix is:
```fortran
type(Variable), pointer :: var
```

--- a/MAPL_Base/MAPL_ExtDataGridCompMod.F90
+++ b/MAPL_Base/MAPL_ExtDataGridCompMod.F90
@@ -2045,7 +2045,7 @@ CONTAINS
         character(len=:), allocatable :: levname
         character(len=:), pointer :: positive 
         type(FileMetadataUtils), pointer :: metadata
-        type(Variable), pointer :: variable
+        type(Variable), pointer :: var
         type(ESMF_TimeInterval)    :: zero
 
         positive=>null()
@@ -2116,16 +2116,16 @@ CONTAINS
 
         end if
 
-        variable => null()
+        var => null()
         if (item%isVector) then
-           variable=>metadata%get_variable(trim(item%fcomp1))
-           _ASSERT(associated(variable),"Variable not found in file")
-           variable => null()
-           variable=>metadata%get_variable(trim(item%fcomp2))
-           _ASSERT(associated(variable),"Variable not found in file")
+           var=>metadata%get_variable(trim(item%fcomp1))
+           _ASSERT(associated(var),"Variable not found in file")
+           var => null()
+           var=>metadata%get_variable(trim(item%fcomp2))
+           _ASSERT(associated(var),"Variable not found in file")
         else
-           variable=>metadata%get_variable(trim(item%var))
-           _ASSERT(associated(variable),"Variable not found in file")
+           var=>metadata%get_variable(trim(item%var))
+           _ASSERT(associated(var),"Variable not found in file")
         end if
     
         levName = metadata%get_level_name(rc=status)

